models/classes/class.QtiTestService.php

```diff
@@ -873,19 +875,46 @@ private function deleteTestsFromClassByLabel(
         core_kernel_classes_Class $itemClass
     ): void {
         $testService = $this->getTestService();
-        $itemTreeService = $this->getItemTreeService();
+
+        $deletedTests = [];
+        $deletedItemClasses = [];
+        $refs = [];
 
         foreach ($testClass->getInstances() as $testInstance) {
             if ($testInstance->getLabel() === $testLabel) {
+                $deletedTests[] = $testInstance->getUri();
```

Comment:
suspicious, shouldnt you add it to list after deletion and not before? :)

Reply:
Changed: b7adabb66

For me it feels a bit weird to delete an object and then retrieve one of its values (for example, if we would be using an ORM with a collection that is lazy-loaded, trying to read it after deletion could break). By the way, in this case the deletion does not modify the object we already have in-memory, so it works fine.

Reply:
Note this has been changed since then to first loop over all item classes and tests to collect their URIs, then to trigger a `TestDeletedEvent` (sic) and finally to do the test and item class deletion per se based on the previously collected URIs.